psi4/driver/qcdb/basislistother.py

```diff
@@ -505,3 +505,56 @@ def load_basfam_other():
     basis_6311ppg_3df_2p_.add_rifit('aug-cc-pvtz-ri')
     basis_6311ppg_3df_2pd_.add_rifit('aug-cc-pvtz-ri')
     basis_6311ppg_3df_3pd_.add_rifit('aug-cc-pvtz-ri')
+
+    # Peterson's nZaPa-NR basis sets
+    basis_2zapa_nr = BasisFamily('3zapa-nr',zeta=2)
+    basis_3zapa_nr = BasisFamily('3zapa-nr',zeta=3)
+    basis_4zapa_nr = BasisFamily('4zapa-nr',zeta=4)
+    basis_5zapa_nr = BasisFamily('5zapa-nr',zeta=5)
+    basis_6zapa_nr = BasisFamily('6zapa-nr',zeta=6)
+    basis_7zapa_nr = BasisFamily('7zapa-nr',zeta=7)
+
+    # unverified fitting sets for nZaPa-NR
+    # Dunnings zeta+1 to be safe, tested on water dimer
+    # the full aug-JKFIT is possibly too much
+    basis_2zapa_nr.add_jkfit('aug-cc-pvtz-jkfit')
+    basis_3zapa_nr.add_jkfit('aug-cc-pvqz-jkfit')
+    basis_4zapa_nr.add_jkfit('aug-cc-pv5z-jkfit')
+    basis_5zapa_nr.add_jkfit('aug-cc-pv5z-jkfit')
+    basis_5zapa_nr.add_jkfit('aug-cc-pv5z-jkfit')
+    basis_6zapa_nr.add_jkfit('aug-cc-pv6z-ri')
+
+    basis_2zapa_nr.add_rifit('aug-cc-pvtz-ri')
+    basis_3zapa_nr.add_rifit('aug-cc-pvqz-ri')
+    basis_4zapa_nr.add_rifit('aug-cc-pv5z-ri')
+    basis_5zapa_nr.add_rifit('aug-cc-pv5z-ri')
+    basis_5zapa_nr.add_rifit('aug-cc-pv5z-ri')
+    basis_6zapa_nr.add_rifit('aug-cc-pv6z-ri')
+
+    basisfamily_list.append(basis_2zapa_nr)
+    basisfamily_list.append(basis_3zapa_nr)
+    basisfamily_list.append(basis_4zapa_nr)
+    basisfamily_list.append(basis_5zapa_nr)
+    basisfamily_list.append(basis_6zapa_nr)
+    basisfamily_list.append(basis_7zapa_nr)
```

Comment:
try running a 7zapa calc density fitted and make sure it throws an error for insufficient jkfit, rather than defaulting to def2-qzvpp, which is the universal default and definitely insufficient.

Reply:
It defaults to def2/J. Is there already somewhere such an error handler?

Reply:
To clarify, the below with 3-zeta will run just fine because even though no JKfit is available for that element and basis, it can use def2-qzvpp (much better periodic table coverage) as a universal default b/c 3<=4. For the 6-zeta, Dunning JKfits similarly unavailable but no suitable def2 parachute, so it throws an error. The logic is happening [here](https://github.com/psi4/psi4/blob/master/psi4/driver/qcdb/libmintsbasisset.py#L812-L813).

I'm just wanting to make sure that [567]zapa are behaving like cc-pv6z and erroring, rather than running with a way-too-small aux basis.

```
molecule {
1 1
Li
}

energy('hf/cc-pvtz')  # fine
energy('hf/cc-pv6z')  # error
```

3-zeta
```
  Basis Set: CC-PVTZ
    Blend: CC-PVTZ
    Number of shells: 10
    Number of basis function: 30
    Number of Cartesian functions: 35
    Spherical Harmonics?: true
    Max angular momentum: 3

   => Loading Basis Set <=

    Name: (CC-PVTZ AUX)
    Role: JKFIT
    Keyword: DF_BASIS_SCF
    atoms 1 entry LI         line    59 file /home/psilocaluser/gits/hrw-release/objdir37/stage/share/psi4/basis/def2-qzvpp-jkfit.gbs
```

6-zeta
```
BasisSetNotFound: BasisSet::construct: Unable to find a basis set for atom 1 for key BASIS among:
  Shell Entries: ['LI']
  Basis Sets: [('CC-PV6Z', 'CC-PV6Z', None)]
  File Path: /home/psilocaluser/gits/hrw-release/objdir37, /home/psilocaluser/gits/hrw-release/objdir37, /home/psilocaluser/gits/hrw-release/objdir37/stage/share/psi4/basis
```